saleor/graphql/product/mutations/products.py

```diff
@@ -471,8 +506,13 @@ class Meta:
         description = "Clears private metadata item for collection."
         permissions = (ProductPermissions.MANAGE_PRODUCTS,)
         public = False
-        error_type_class = ProductError
-        error_type_field = "product_errors"
+        error_type_class = CollectionError
+        error_type_field = "collection_errors"
+
+    @classmethod
+    def success_response(cls, instance):
+        instance = ChannelContext(node=instance, channel_slug=None)
+        return super().success_response(instance)
 
 
 class CategoryUpdateMeta(UpdateMetaBaseMutation):
```

Comment:
Still unresolved:
"All mutations that return collection should wrap it in ChannelContext.
You should add it to all deprecated collection meta mutations in this file."